src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test/Completion/RazorDirectiveAttributeCompletionSourceTest.cs

```diff
@@ -73,6 +73,27 @@ public void InitializeCompletion_PageDirective_ReturnsDoesNotParticipate()
             Assert.Equal(expectedApplicableToSpan, result.ApplicableToSpan);
         }
 
+        [Theory]
+        [InlineData("@Value[0]")]
+        [InlineData("@DateTime.Now")]
+        [InlineData("@SomeMethod()")]
+        [InlineData("@(DateTime.Now)")]
+        [InlineData("@{SomeProperty;}")]
+        public void InitializeCompletion_ComplexExpressions_ReturnsDoesNotParticipate(string expression)
```

Comment:
Are these the tests for the new *invalid characters* check?

Reply:
Yes